akka-cluster/src/main/scala/akka/cluster/sbr/DowningStrategy.scala

```diff
@@ -281,30 +282,35 @@ import akka.coordination.lease.scaladsl.Lease
     }
   }
 
-  private def additionalNodesToDownWhenIndirectlyConnected: Set[UniqueAddress] = {
+  private def additionalNodesToDownWhenIndirectlyConnected(downable: Set[UniqueAddress]): Set[UniqueAddress] = {
     if (unreachableButNotIndirectlyConnected.isEmpty)
       Set.empty
     else {
       val originalUnreachable = _unreachable
       val originalReachability = _reachability
+
       try {
         val intersectionOfObserversAndSubjects = indirectlyConnectedFromIntersectionOfObserversAndSubjects
         val haveSeenCurrentGossip = indirectlyConnectedFromSeenCurrentGossip
-        // remove records between the indirectly connected
-        _reachability = reachability.filterRecords(
-          r =>
-            !((intersectionOfObserversAndSubjects(r.observer) && intersectionOfObserversAndSubjects(r.subject)) ||
-            (haveSeenCurrentGossip(r.observer) && haveSeenCurrentGossip(r.subject))))
+        _reachability = reachability
+          .filterRecords { r =>
+            // we only retain records for addresses that are still downable
+            downable.contains(r.observer) && downable.contains(r.subject) &&
+            // remove records between the indirectly connected
+            !(intersectionOfObserversAndSubjects(r.observer) && intersectionOfObserversAndSubjects(r.subject) ||
+              haveSeenCurrentGossip(r.observer) && haveSeenCurrentGossip(r.subject))
+          }
         _unreachable = reachability.allUnreachableOrTerminated
-        val additionalDecision = decide()
 
+        val additionalDecision = decide()
         if (additionalDecision.isIndirectlyConnected)
           throw new IllegalStateException(
             s"SBR double $additionalDecision decision, downing all instead. " +
             s"originalReachability: [$originalReachability], filtered reachability [$reachability], " +
             s"still indirectlyConnected: [$indirectlyConnected], seenBy: [$seenBy]")
```

Comment:
I wonder if it really needed to down all in such a situation. 
Wouldn't that be enough to only down the nodes we already detected as indirectly connected? 

Like in:

```scala
if (additionalDecision.isIndirectlyConnected) Set.empty
else nodesToDown(additionalDecision)
```


As such, `additionalNodesToDownWhenIndirectlyConnected` will be just trying to find additional nodes to down, but if we get stuck with the already indirectly connected nodes, we just down those instead of a down all. 

Does that make sense? Am I missing a corner case that would justify a DownAll?

Reply:
Hmm, I think the risk in just removing the indirectly connected is that we may have two sides of the split that assumes it's in the majority because none of the sides were able to get the full picture of which nodes to down.

It that's the case, keep only the downable node in the reachability table is a safer option.

Reply:
hi,We have encountered this problem too, when will it be solved completely?